inference-engine/src/mkldnn_plugin/nodes/mkldnn_pooling_node.cpp

```diff
@@ -178,14 +185,70 @@ void MKLDNNPoolingNode::getSupportedDescriptors() {
     }
 }
 
-void MKLDNNPoolingNode::createPrimitive() {
-    if (prim)
-        return;
+std::pair<std::vector<ptrdiff_t>, std::vector<ptrdiff_t>> MKLDNNPoolingNode::getPaddingFromNode(std::shared_ptr<ngraph::Node> node) const {
+    const auto convertPadding = [](const VectorDims &newPads) {
+        std::vector<ptrdiff_t> pads(newPads.size());
+        for (int i = 0; i < newPads.size(); i++) {
+            pads[i] = static_cast<ptrdiff_t>(newPads[i]);
+        }
+        return pads;
+    };
 
-    mkldnn::primitive_attr attr;
-    setPostOps(attr, true);
+    VectorDims padsBegin, padsEnd;
+    if (getAlgorithm() == PoolingMax) {
+        const auto pool = ngraph::as_type_ptr<const ngraph::op::v1::MaxPool>(opToShapeInfer);
+        padsBegin = pool->get_pads_begin();
+        padsEnd = pool->get_pads_end();
+    } else if (getAlgorithm() == PoolingAvg) {
+        const auto pool = ngraph::as_type_ptr<const ngraph::op::v1::AvgPool>(opToShapeInfer);
+        padsBegin = pool->get_pads_begin();
+        padsEnd = pool->get_pads_end();
+    }
+    return {convertPadding(padsBegin), convertPadding(padsEnd)};
+}
+
+void MKLDNNPoolingNode::prepareParams() {
+    const NodeDesc *selected_pd = getSelectedPrimitiveDescriptor();
+    if (selected_pd == nullptr)
+        IE_THROW()  << "Pooling node with name '" << getName() << "' did not set preferable primitive descriptor";
+
+    AttrPtr attr;
+    if (isDynamicNode()) {
+        if (!pAttr) {
+            pAttr = initPrimitiveAttr();
+        }
+        attr = pAttr;
+    } else {
+        attr = initPrimitiveAttr();
+    }
+
+    auto inDesc = getParentEdgesAtPort(0)[0]->getMemory().GetDescWithType<DnnlMemoryDesc>();
+    const auto& in_candidate = inDesc->getDnnlDesc();
+    auto outDesc = getChildEdgesAtPort(0)[0]->getMemory().GetDescWithType<DnnlMemoryDesc>();
+    const auto& out_candidate = outDesc->getDnnlDesc();
 
-    auto prim_desc = createPrimitiveDescriptor<pooling_forward::primitive_desc, pooling_forward::desc>(attr);
+    if (isDynamicNode()) {
+        if (auto_pad) {
+            std::tie(data_pad_begin, data_pad_end) = getPaddingFromNode(opToShapeInfer);
```

Comment:
minor: data_pad_begin/end are class fields. Why not to update them inside `getPaddingFromNode` in order to avoid make_pair/tie overheads?
Also it looks like we can combine `initEffectivePad` and `getPaddingFromNode` inside one method.